straxen/plugins/acqmon_processing.py

```diff
@@ -216,3 +216,125 @@ def compute(self, events, aqmon_hits):
+                  ]
+        return dtype
+
+    def compute(self, raw_records_aqmon,
+                raw_records_aqmon_nv,
+                raw_records_aux_mv,
+                start, end):
+        rr_tpc = raw_records_aqmon
+        rr_nv = raw_records_aqmon_nv
+        rr_mv = raw_records_aux_mv
+
+        extra_offset = 0
+        _mask_tpc = (rr_tpc['channel'] == 798)
+        if not np.any(_mask_tpc):
+            # For some runs in the beginning no signal has been acquired here.
+            # In that case we have to add the internal DAQ delay as an extra offset later.
+            _mask_tpc = (rr_tpc['channel'] == 801)
+            extra_offset = self.config['tpc_internal_delay'][self.run_id >= DIGITIZER_DELAY_CHANGE_RUNID]
+
+        hits_tpc = self.get_nim_edge(rr_tpc[_mask_tpc], self.config['adc_threshold_nim_signal'])
+
+        _mask_mveto = (rr_mv['channel'] == 1084)
+        hits_mv = self.get_nim_edge(rr_mv[_mask_mveto], self.config['adc_threshold_nim_signal'])
+
+        _mask_nveto = rr_nv['channel'] == 813
+        hits_nv = self.get_nim_edge(rr_nv[_mask_nveto], self.config['adc_threshold_nim_signal'])
+        nveto_extra_offset = 0
+        if not len(hits_nv):
+            # During SR0 sync signal was not recorded properly for the
+            # neutron-veto, hence take waveform itself as "hits".
+            _mask_nveto &= rr_nv['record_i'] == 0
+            nveto_extra_offset = self.config['epsilon_offset']
+            hits_nv = rr_nv[_mask_nveto]
+
+        offsets_mv = self.estimate_delay(hits_tpc, hits_mv)
+        offsets_nv = self.estimate_delay(hits_tpc, hits_nv)
+        assert len(offsets_mv) == len(offsets_nv), 'Unequal number of sync signals!'
+
+        result = np.zeros(len(offsets_mv), dtype=self.dtype)
+        result['time'] = hits_tpc['time']
+        result['endtime'] = strax.endtime(hits_tpc)
+        result['time_offset_nv'] = offsets_nv + extra_offset + nveto_extra_offset
+        result['time_offset_mv'] = offsets_mv + extra_offset
+
+        return result
+
+    @staticmethod
+    def get_nim_edge(raw_records, threshold=500):
+        records = strax.raw_to_records(raw_records)
+        strax.baseline(records)
+        hits = strax.find_hits(records, min_amplitude=threshold)
+        return hits
+
+    def estimate_delay(self, hits_det0, hits_det1):
+        """
+        Function to estimate the average offset between two hits.
+        """
+        err_value = -10000000000
+        coinc_time = TO_BE_EXCPECTED_MAX_DELAY
+
+        offsets = []
+        count_valid = 0
+        prev_time = 0
+        for i in range(len(hits_det0)):
+            offset = self.find_offset_nearest(hits_det1['time'], hits_det0['time'][i])
+            time_to_prev = hits_det0['time'][i] - prev_time
+
+            # Additional check to avoid spurious signals
+            if (abs(offset) < coinc_time) & (time_to_prev > 1e10):
+                offsets.append(offset)
+                count_valid += 1
```

Comment:
Do you mean the count_valid? We should check if it's >0. If it's not the case (which points to problems in recording the synchronization signal) we should issue an error (it should never happens, but better to be safe).

Reply:
Ah yes sorry, I changed this a bit Andrea. The plugin will compute the offsets for all Clock signals in a run. The averaging can be then done later. Hence, the counter is not needed anymore. In case we do not have any valid signal I just add the error value.